pennylane/ops/op_math/adjoint_class.py

```diff
@@ -224,3 +232,21 @@ def diagonalizing_gates(self):
 
     def adjoint(self):
         return self.base.queue()
+
+    @property
+    def arithmetic_depth(self) -> int:
+        return 1 + self.base.arithmetic_depth
+
+    def simplify(self, depth=-1):
+        # TODO: Should we add a self.do_queue attribute and instantiate the simplified class
+        # with the same value?
+        if depth == 0:  # TODO: We could add a decorator to do this check for all simplify methods
+            return self
+        # TODO: Should we do: Adj(AB).simplify() = Adj(B) Ajd(A) ?
+        if isinstance(self.base, Adjoint) and (depth > 1 or depth < 0):  # Adj(Adj(A)) = A
```

Comment:
```suggestion
        if isinstance(self.base, Adjoint):  # Adj(Adj(A)) = A
```
The `if depth==0` takes care of the second check.
If `depth==1`, then the base can't be an `Adjoint`.